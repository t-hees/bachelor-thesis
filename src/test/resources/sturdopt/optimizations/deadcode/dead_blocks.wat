(module
  (type $add_type (func (param i32) (param i32) (result i32)))
  (type (;1;) (func (param i32) (result i32))) ;; Type created by loop
  (type $result_type (func (result i32)))
  (func $add (type $add_type)
    local.get 0
    (loop $loop_without_branch (param i32) (result i32)
      local.get 1
      i32.add
    )
  )
  (func (export "main") (type $result_type) (result i32)
    (block $block_without_branch (result i32)
      i32.const 0
      i32.const 0
      br_if $block_without_branch
      drop
      i32.const 2
    )
    i32.const 3
    call $add
    return
    (block $unused_block (result i32)
      i32.const 1
    )
  )
)
