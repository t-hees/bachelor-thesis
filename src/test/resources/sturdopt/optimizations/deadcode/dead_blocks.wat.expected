(module
  (type $add_type (func (param i32) (param i32) (result i32)))
  (type (;1;) (func (param i32) (result i32))) ;; Type created by loop
  (type $result_type (func (result i32)))
  (func $add (type $add_type)
    local.get 0
    local.get 1
    i32.add
  )
  (func (export "main") (type $result_type) (result i32)
    i32.const 0
    i32.const 0
    drop
    drop
    i32.const 2
    i32.const 3
    call $add
    return
    unreachable
  )
  (func (export "infinite loop")
    (loop
      i32.const 1
      br_if 0
    )
  )
)
